sdk/core/azure-core/src/http/body_stream.cpp

```diff
@@ -103,10 +104,12 @@ int64_t FileBodyStream::Read(Azure::Core::Context const& context, uint8_t* buffe
   return result;
 }
 #elif defined(AZ_PLATFORM_WINDOWS)
-int64_t FileBodyStream::Read(Azure::Core::Context const& context, uint8_t* buffer, int64_t count)
+int64_t FileBodyStream::ReadImplementation(
+    Azure::Core::Context const& context,
+    uint8_t* buffer,
+    int64_t count)
 {
-  context.ThrowIfCanceled();
-
+  (void)context;
```

Comment:
Probably out of scope for this PR:
Should this try to do async i/o rather than sync and honor context cancellation mid-way, using `CancelIoEx`? And possibly use `ReadFileEx` for async read?
https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-readfile#remarks

Reply:
That's a good improvement we can add in the future if someone ask for it right?
Keep in mind that then the SDK is reading a bodyStream for sending a request, we will call `Read()` multiple times in chunks, so we are already kind of asking if the customer wants to cancel the request while a file is been read.

Adding the cacellation check for FileBody implementation would empower the end-user to cancel the read while the bodyStream is used out of an HTTP request with the SDK